src/include/mpi.h.in

```diff
@@ -910,6 +910,12 @@ typedef int (MPI_Datarep_extent_function)(MPI_Datatype datatype, MPI_Aint *,
                       void *);
 #define MPI_CONVERSION_FN_NULL ((MPI_Datarep_conversion_function *)0)
 
+typedef struct {
+    void **storage_stack;
+} QMPI_Context;
+
+#define QMPI_MAX_TOOL_NAME_LENGTH 256
+
```

Comment:
I am thinking maybe we can add a macros `#define HAS_QMPI 1` -- make it cofigure macro, then the tests can simply guard with `#ifdef HAS_QMPI`. That way the tests gets run when qmpi is available and reduce to a null-test when it isn't.

Having feature test macros are generally useful, but admittedly non-standard. I think it is ok to have some implementation-specific macros (which we already do) in `mpi.h`, what do you think?

Reply:
I think it's fine to put whatever we want in `mpi.h`. The contents aren't mandated by the MPI Standard. We previously had all of our `MPIX` definitions in here.

That being said, the reason I didn't put the macros here was because of the tests. The test needed to be able to compile all of the time, even if there was no tool attached.

One thing I forgot to add is a way to skip/abort the test if there are no tools attached.